Source/Library/Adapt.Analytics/AngleCalculationAnalytic.cs

```diff
@@ -0,0 +1,108 @@
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  08/02/2021 - C. Lackner
+//       Generated original version of source code.
+//  12/08/2021 - A. Hagemeyer
+//       Changed to angle calculation analytic
+//
+// ******************************************************************************************************
+
+
+using Adapt.Models;
+using GemstoneCommon;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Numerics;
+using System.Runtime;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace Adapt.DataSources
+{
+    /// <summary>
+    /// Caculates the angle given magnitude and phase
+    /// </summary>
+
+    [AnalyticSection(AnalyticSection.DataCleanup)]
+
+    [Description("Angle Calculation: Calculate the angle given magnitude and phase")]
+    public class AngleCalculation : IAnalytic
+    {
+        private Setting m_settings;
+        private int m_fps;
+        public class Setting
+        {
+            public AngleUnit Unit { get; set; }
+        }
+
+        public Type SettingType => typeof(Setting);
+
+        public int FramesPerSecond => m_fps;
+
+        int IAnalytic.PrevFrames => 0;
+
+        int IAnalytic.FutureFrames => 0;
+
+        public IEnumerable<string> OutputNames()
+        {
+            return new List<string>() { "Angle" };
+        }
+
+        public IEnumerable<string> InputNames()
+        {
+            return new List<string>() { "Magnitude", "Phase"};
+        }
+
+        public Task<ITimeSeriesValue[]> Run(IFrame frame, IFrame[] previousFrames, IFrame[] futureFrames)
+        {
+            AdaptValue result = new AdaptValue("Angle", getAngle(frame), frame.Timestamp);
+            return Task.FromResult<ITimeSeriesValue[]>(new AdaptValue[] { result });
+        }
+
+        public double getAngle(IFrame frame) 
+        {
+            ITimeSeriesValue magnitude = frame.Measurements["Magnitude"];
+            ITimeSeriesValue phase = frame.Measurements["Phase"];
+            if (m_settings.Unit == AngleUnit.Degrees)
+                return Math.Atan(magnitude.Value * Math.Sin(phase.Value) / magnitude.Value * Math.Cos(phase.Value));
+            else
+                return Math.Atan(magnitude.Value * Math.Sin((180 / Math.PI) * phase.Value) / magnitude.Value * Math.Cos((180 / Math.PI) * phase.Value));
+        }
+
+        public void Configure(IConfiguration config)
+        {
+            m_settings = new Setting();
+            config.Bind(m_settings);
+        }
+
+        public void SetInputFPS(IEnumerable<int> inputFramesPerSecond) 
+        {
+            m_fps = inputFramesPerSecond.FirstOrDefault();
```

Comment:
Change to least Common Multiple. This approach may cause issues when looking at 20 and 30 fps signals.